### DISCUSSION TITLE
OPA server - prometheus histogram buckets configuration

### DISCUSSION THREAD

**Discussion by AdrianArnautu** [open-policy-agent/community#487] [OPA and Rego] [answered]
Hi folks,

OPA exposes a prometheus histogram ```http_request_duration_seconds``` for the request duration, with only a handful of buckets.
From https://github.com/open-policy-agent/opa/blob/main/internal/prometheus/prometheus.go#L47 and from the ```:8181/metrics```

```
1e-6, // 1 microsecond
5e-6,
1e-5,
5e-5,
1e-4,
5e-4,
1e-3, // 1 millisecond
0.01,
0.1,
1, // 1 second
```

Is there any way the buckets can be set from the config? Did I missed that out from the documentation/code?

**Comment by srenatus** [comment, marked as answer]
Right now, bucket sizes are not configurable, sorry.

**Comment by charlieegan3** [comment]
If you'd like to open a new issue for this with some details about your use case that'd be interesting. 

If it's a clear use case that other users have in common we could certainly consider it.

**Comment by AdrianArnautu** [comment]
thank you guys
we might lend a hand here in the near future, we need a better insight on what's going on at OPA level

**Reply by charlieegan3** [reply]
Happy to help in slack if you need help debugging a policy performance issue 🙂 (tomorrow!)

**Comment by AdrianArnautu** [comment]
I've raised https://github.com/open-policy-agent/opa/issues/6238 that adds support for configuration
